poetry/poetry.py

```diff
@@ -147,7 +148,7 @@ def create(cls, cwd):  # type: (Path) -> Poetry
 
                 package.add_dependency(name, constraint)
 
-        if "dev-dependencies" in local_config:
+        if "dev-dependencies" in local_config and not "--no-dev" in sys.argv:
```

Comment:
Hm this change may affect all commands ( did not test it yet )

Reply:
Indeed, this doesn't look like the correct place to manage the behavior of the `install` command. What the `--no-dev` option does is it sets `Installer.dev_mode` attribute and you can probably work off of that.

Reply:
Hello,
yes you are right, this is a bad place, its more like patch / workaround. Unfortunately if you look closely into the code, you would find that class Poetry defined in this file is initialised in every single command, even the one which does not to anything effective, therefore it is required to change this control mechanism somehow
JJ.